modules/docker/inspect.go

```diff
@@ -0,0 +1,204 @@
+	"testing"
+	"time"
+)
+
+// ContainerInspect defines the output of the Inspect method, with the options returned by 'docker inspect'
+// converted into a more friendly and testable interface
+type ContainerInspect struct {
+	// ID of the inspected container
+	ID       string
+
+	// Name of the inspected container
+	Name     string
+
+	// time.Time that the container was created
+	Created  time.Time
+
+	// String representing the container's status
+	Status   string
+
+	// Whether the container is currently running or not
+	Running  bool
+
+	// Container's exit code
+	ExitCode uint8
+
+	// String with the container's error message, if there is any
+	Error    string
+
+	// Ports exposed by the container
+	Ports    []Port
+
+	// Volume bindings made to the container
+	Binds    []VolumeBind
+}
+
+// Port represents a single port mapping exported by the container
+type Port struct {
+	HostPort      uint16
+	ContainerPort uint16
+	Protocol      string
+}
+
+// VolumeBind represents a single volume binding made to the container
+type VolumeBind struct {
+	Source      string
+	Destination string
+}
+
+// inspectOutput defines options that will be returned by 'docker inspect', in JSON format.
+// Not all options are included here, only the ones that we might need
+type inspectOutput struct {
+	Id      string
+	Created string
+	Name    string
+	State   struct {
+		Status   string
+		Running  bool
+		ExitCode uint8
+		Error    string
+	}
+	NetworkSettings struct {
+		Ports map[string][]struct {
+			HostIp   string
+			HostPort string
+		}
+	}
+	HostConfig struct {
+		Binds []string
+	}
+}
+
+func Inspect(t *testing.T, id string) ContainerInspect {
+	out, err := InspectE(t, id)
+	require.NoError(t, err)
+	return out
+}
+
+// InspectE runs the 'docker inspect {container id}' command and returns a ContainerInspect
+// struct, converted from the output JSON, along with any errors
+func InspectE(t *testing.T, id string) (ContainerInspect, error) {
+	match, _ := regexp.MatchString("[[:alnum:]]", id)
+	if !match {
+		return ContainerInspect{}, fmt.Errorf("%s is not a valid container ID", id)
+	}
+
+	cmd := shell.Command{
+		Command: "docker",
+		Args:    []string{"container", "inspect", id},
+	}
+
+	out, err := shell.RunCommandAndGetOutputE(t, cmd)
+	if err != nil {
+		return ContainerInspect{}, err
+	}
+
+	var containers []inspectOutput
+	err = json.Unmarshal([]byte(out), &containers)
+	if err != nil {
+		return ContainerInspect{}, err
+	}
+
+	if len(containers) == 0 {
+		return ContainerInspect{}, nil
+	}
+
+	c := containers[0]
+
+	return transformContainer(t, c), nil
+}
+
+// transformContainerPorts converts 'docker inspect' output JSON into a more friendly and testable format
+func transformContainer(t *testing.T, c inspectOutput) ContainerInspect {
+	name := strings.TrimLeft(c.Name, "/")
+	ports := transformContainerPorts(t, c)
+	volumes := transformContainerVolumes(t, c)
+
+	created, err := time.Parse(time.RFC3339Nano, c.Created)
+	require.NoError(t, err)
```

Comment:
In order to fulfill the contract of `E` functions (e.g., `inspectE`), this function should return the error rather than calling `require.NoError`.